controllers/runner_controller.go

```diff
@@ -259,6 +321,27 @@ func (r *RunnerReconciler) processRunnerCreation(ctx context.Context, runner v1a
 	return ctrl.Result{}, nil
 }
 
+func (r *RunnerReconciler) createObject(ctx context.Context, obj client.Object, meta metav1.ObjectMeta, runner v1alpha1.Runner, owner metav1.Object, log logr.Logger) *ctrl.Result {
```

Comment:
I believe the `owner` can be merged into `runner` if we received it as `*v1alpha1.Runner`

```suggestion
func (r *RunnerReconciler) createObject(ctx context.Context, obj client.Object, meta metav1.ObjectMeta, runner *v1alpha1.Runner, log logr.Logger) *ctrl.Result {
```